routingv8/request.go

```diff
@@ -527,3 +529,54 @@ type RouteImportRequest struct {
 type RouteImportRequestBody struct {
 	Trace []GeoWaypoint `json:"trace"`
 }
+
+type Vehicle struct {
+	// Gross vehicle weight, including trailers and shipped goods when loaded at capacity, specified in kilograms.
+	// If unspecified, it will default to currentWeight.
+	// If neither parameter has a value specified, it will default to 0.
+	// Notes:
+	// - Supported in truck, bus, privateBus, car (Beta), taxi (Beta) transport modes.
+	// - Maximum weight for a car or taxi without a trailer is 4250 kg.
+	// - Maximum weight for a car or taxi with a trailer is 7550 kg.
+	GrossWeight int
```

Comment:
Can we document that it should be a negative value? (Maybe it is worth using a `uint` instead?)

Reply:
All documentation here is just copy-paste from HERE. How much client-side validation should we do? Isn't it enough that HERE returns an error if the values are wrong?

Reply:
Yeah I think that's how we are doing it and might as well keep doing it. The ranges are all taken from HERE documentation as well so I thought it would be good since we are already documenting everything else. The `uint` part we can skip. Was just a nice QoL change

Reply:
True. Since it's also in HERE docs I will add it here as well. I'll skip `uint` for now. Feels like we're not using that type much in other places and it would just be cumbersome to convert, IMO.

Reply:
Updated with the ranges where applicable.